Data azurerm_express_route_circuit_peering fails to be read with "cannot parse an empty string"

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.8.5

### AzureRM Provider Version

3.108.0

### Affected Resource(s)/Data Source(s)

azurerm_express_route_circuit_peering

### Terraform Configuration Files

```hcl
data "azurerm_express_route_circuit_peering" "example" {
  peering_type               = "AzurePrivatePeering"
  express_route_circuit_name = "example"
  resource_group_name        = "example"
}
```


### Debug Output/Panic Output

```shell
Sanitized output of the relevant debug logs:


module.vwan.data.azurerm_express_route_circuit_peering.example[0]: Reading...
[ERROR] provider.terraform-provider-azurerm_v3.108.0_x5: Response contains error diagnostic: diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="parsing \"\": cannot parse an empty string" tf_data_source_type=azurerm_express_route_circuit_peering tf_proto_version=5.4 tf_provider_addr=provider tf_req_id=1f0cff07-7645-bda0-ec0f-ca4a73ac68d7 tf_rpc=ReadDataSource @caller=github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto timestamp=2024-06-21T02:11:28.893Z
[ERROR] vertex "module.vwan.data.azurerm_express_route_circuit_peering.example[0]" error: parsing "": cannot parse an empty string
[ERROR] vertex "module.vwan.data.azurerm_express_route_circuit_peering.example (expand)" error: parsing "": cannot parse an empty string
Error: parsing "": cannot parse an empty string

with module.vwan.data.azurerm_express_route_circuit_peering.example[0],
on modules/vwan_hub/vwan.tf line 52, in data "azurerm_express_route_circuit_peering" "example":
52: data "azurerm_express_route_circuit_peering" "example" {
```


### Expected Behaviour

The data source of `azurerm_express_route_circuit_peering` doesn't return an error when used.

### Actual Behaviour

The data source of `azurerm_express_route_circuit_peering` returns an error when used.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/express_route_circuit_peering_data_source.go
+++ b/internal/services/network/express_route_circuit_peering_data_source.go
@@ -106,21 +106,20 @@ func dataSourceExpressRouteCircuitPeering() *pluginsdk.Resource {
 func dataSourceExpressRouteCircuitPeeringRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Network.ExpressRouteCircuitPeerings
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	defer cancel()
 
-	// TODO this should be constructed?
-	id, err := commonids.ParseExpressRouteCircuitPeeringID(d.Id())
-	if err != nil {
-		return err
-	}
+	id := commonids.NewExpressRouteCircuitPeeringID(subscriptionId, d.Get("resource_group_name").(string), d.Get("express_route_circuit_name").(string), d.Get("peering_type").(string))
 
-	resp, err := client.Get(ctx, *id)
+	resp, err := client.Get(ctx, id)
 	if err != nil {
 		if response.WasNotFound(resp.HttpResponse) {
 			return fmt.Errorf("%s was not found", id)
 		}
-		return fmt.Errorf("retrieving %s: %+v", *id, err)
+		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
+
+	d.SetId(id.ID())
 
 	d.Set("peering_type", id.PeeringName)
 	d.Set("express_route_circuit_name", id.CircuitName)

